daft/series.py

```diff
@@ -35,7 +35,10 @@ def _from_pyseries(pyseries: PySeries) -> Series:
 
     @staticmethod
     def from_arrow(array: pa.Array | pa.ChunkedArray, name: str = "arrow_series") -> Series:
-        if isinstance(array, pa.Array):
+        if DataType.from_arrow_type(array.type) == DataType.python():
+            # If the Arrow type is not natively supported, go through the Python list path.
```

Comment:
Hmm not sure if logging a warning here would be the best UX, it could be quite noisy and this is our current designated path for supporting these types. Even with a local log-once, this would still log a warning in every read task.

Are we currently logging warnings in other Python object fallback areas?

Reply:
Added a debug logging statement for now, happy to change it to a warning if we don't think it will be too spammy.